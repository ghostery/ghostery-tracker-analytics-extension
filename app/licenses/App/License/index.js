import License from './License.jsx';

export default License;
