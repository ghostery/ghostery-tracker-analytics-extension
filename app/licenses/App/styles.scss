/*!
 * Licenses CSS
 *
 * Insights by Ghostery
 * https://www.ghostery.com/insights/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 * See https://www.ghostery.com/eula for license agreement.
 */
@import '../../utils/scss/overrides.scss';
@import '../../utils/scss/open-sans.scss';
@import '../../utils/scss/color-variables.scss';

$header-offset: 64px;
$footer-offset: 40px;

.Licenses {
  background-image: $background-gradient;
  background-repeat: no-repeat;
  width: 100%;
  height: 100%;
  font-family: OpenSans;
  font-weight: 400;
  background-attachment: fixed;
}

.Licenses__header {
  border-bottom: 1px solid  $medium-grey;
  width: 100%;
  height: $header-offset;
}

.Licenses__logo {
  margin: 17px 0 0 22px;
  padding: 0;
}

.Licenses__title {
  margin: 19px 55px 0 0;
  font-size: 20px;
}

.Licenses__list {
  overflow-y: auto;
  height: calc(100vh - #{$header-offset} - #{$footer-offset});
}

.Licenses__copyright {
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: $darker-grey;
  height: $footer-offset;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 10px;
}
