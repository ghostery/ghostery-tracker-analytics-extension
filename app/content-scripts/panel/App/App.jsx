import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../../../vendor/bootstrap.scss';

import Header from '../../../shared-components/Header';
import PageStats from '../../../shared-components/PageStats';
import TrackerList from '../../../shared-components/TrackerList';
import TrackerDistributionGraph from '../../../shared-components/TrackerDistributionGraph';
import PageLatencyGraph from '../../../shared-components/PageLatencyGraph';
import PageSizeGraph from '../../../shared-components/PageSizeGraph';
import ThemedToast from '../../../shared-components/ThemedToast';
import './styles.scss';

import onMessage from '../../../store/messages/messageListeners';


class App extends Component {
  constructor(props) {
    super(props);
    this.panelRef = React.createRef();
  }

  componentDidMount() {
    if (!window.port) { window.port = chrome.runtime.connect({ name: 'window' }); }
    window.port.onMessage.addListener(this.onPanelMessage);
    window.port.postMessage({ type: 'RequestUserInfo' });
    window.port.postMessage({ type: 'RequestSettings' });
    window.port.postMessage({ type: 'RequestTabDetails' });

    const { actions, messageCreators } = this.props;
    actions.updateTooltipRef(this.panelRef.current);
    messageCreators.sendMetrics({ type: 'visit_insightsPanel' });
  }

  onPanelMessage = (message) => {
    if (message.type === 'RemovePanel') {
      window.port.onMessage.removeListener(this.onPanelMessage);
      const panelEl = document.getElementById('ghostery-insights');
      panelEl.style.opacity = 0;
      setTimeout(() => panelEl.parentNode.removeChild(panelEl), 300);
      return;
    }

    onMessage.bind(this)(message);
  }

  handleClick = () => {
    window.port.postMessage({ type: 'OpenApp' });
  }

  render() {
    const { handleClick } = this;
    return (
      <React.Fragment>
        <div className="Panel__container d-flex justify-content-center">
          <div className="Panel__content" ref={this.panelRef}>
            <Header panel>
              <PageStats panel />
            </Header>
            <div className="Panel__graphContainer d-flex justify-content-center align-items-start">
              <TrackerList handleClick={handleClick} />
              <TrackerDistributionGraph
                height={190}
                width={220}
                handleClick={handleClick}
                expanded={false}
                small
              />
              <PageLatencyGraph height={190} width={220} handleClick={handleClick} small />
              <PageSizeGraph height={190} width={220} handleClick={handleClick} small />
            </div>
            <div className="Graph__tooltip absolute" />
            <div className="Graph__tooltipCaret absolute" />
            <ThemedToast panel />
          </div>
        </div>
      </React.Fragment>
    );
  }
}

App.propTypes = {
  actions: PropTypes.shape({
    setInitialTabInfo: PropTypes.func.isRequired,
    updatePageSize: PropTypes.func.isRequired,
    updateTrackerInfo: PropTypes.func.isRequired,
    updateTimingPerformance: PropTypes.func.isRequired,
    updateIsLive: PropTypes.func.isRequired,
    updateTooltipRef: PropTypes.func.isRequired,
  }).isRequired,
  messageCreators: PropTypes.shape({
    sendMetrics: PropTypes.func.isRequired,
  }).isRequired,
};

export default App;
