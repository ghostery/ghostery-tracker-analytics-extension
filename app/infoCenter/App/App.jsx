import React, { Component } from 'react';
import ClassNames from 'classnames';
import Navbar from 'react-bootstrap/Navbar';
import '../../vendor/bootstrap.scss';

import ProductTour from './ProductTour';
import Glossary from './Glossary';
import ThemedTabs from '../../shared-components/ThemedTabs';
import './styles.scss';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showBackgroundCurve: true,
      midProductTour: false,
    };
  }

  toggleBackgroundCurve = (showBackgroundCurve, midProductTour) => {
    if (typeof midProductTour === 'undefined') {
      this.setState({ showBackgroundCurve });
      return;
    }
    this.setState({ showBackgroundCurve, midProductTour });
  }

  tabSelectHandler = (tab) => {
    const { midProductTour } = this.state;
    if (tab === '1' && midProductTour) {
      this.toggleBackgroundCurve(false);
      return;
    }
    this.toggleBackgroundCurve(true);
  }

  render() {
    const { showBackgroundCurve } = this.state;
    const backgroundCurveClasses = ClassNames('InfoCenter__backgroundCurve', {
      show: showBackgroundCurve,
    });

    return (
      <div
        className="InfoCenter__container d-flex justify-content-center"
        ref={(node) => { this.tooltipRef = node; }}
      >
        <img
          src="/dist/images/info-center/background-curve.svg"
          className={backgroundCurveClasses}
          alt="Background"
        />
        <div className="InfoCenter__content d-flex">
          <Navbar.Brand
            className="InfoCenter__logo"
            href="https://www.ghostery.com/products/"
            target="_blank"
            rel="noopener noreferrer"
          >
            <img
              alt="Insights by Ghostery Icon"
              src="/dist/images/logo-icons/insights-white-text-beta.svg"
            />
          </Navbar.Brand>
          <ThemedTabs
            altTabs
            passedSelectHandler={this.tabSelectHandler}
            stickyNavItems
            tabs={[
              {
                title: 'Product Tour',
                view: <ProductTour toggleBackgroundCurve={this.toggleBackgroundCurve} />,
              },
              { title: 'Glossary', view: <Glossary /> },
            ]}
          />
        </div>
      </div>
    );
  }
}

export default App;
