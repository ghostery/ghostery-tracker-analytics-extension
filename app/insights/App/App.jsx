import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import '../../vendor/bootstrap.scss';

import Header from '../../shared-components/Header';
import PageStats from '../../shared-components/PageStats';
import ThemedTabs from '../../shared-components/ThemedTabs';
import ThemedToast from '../../shared-components/ThemedToast';
import PageNotScannedOverlay from './PageNotScannedOverlay';
import OverviewTab from './OverviewTab';
import GlobalTrendsTab from './GlobalTrendsTab';
import TrackerParentageTab from './TrackerParentageTab';
import './styles.scss';

import onMessage from '../../store/messages/messageListeners';

class App extends Component {
  constructor(props) {
    super(props);
    window.port = chrome.runtime.connect({ name: 'app' });
    window.port.onMessage.addListener(onMessage.bind(this));
  }

  componentDidMount() {
    window.port.postMessage({ type: 'RequestUserInfo' });
    window.port.postMessage({ type: 'RequestSettings' });
    window.port.postMessage({ type: 'RequestTabDetails' });

    const { actions, messageCreators } = this.props;
    actions.updateTooltipRef(this.tooltipRef);
    messageCreators.sendMetrics({ type: 'visit_insightsTab' });
    if (window.location.hash.length === 0) {
      window.location.hash = 'tab=1&nestedTab=1';
    }
  }

  handleClick = () => {
    window.port.postMessage({ type: 'FocusParentTab' });
  }

  sendTrackerInformationTabMetrics = (activeKey) => {
    const { messageCreators } = this.props;
    if (activeKey === '2') {
      messageCreators.sendMetrics({ type: 'visit_trackerInformationTab' });
    }
  }

  render() {
    const { parentTabUrl } = this.props;

    const containerClassNames = ClassNames(
      'App__container',
      'd-flex',
      'justify-content-center',
    );

    return (
      <React.Fragment>
        <PageNotScannedOverlay />
        <div
          className={containerClassNames}
          ref={(node) => { this.containerRef = node; }}
        >
          <div className="App" ref={(node) => { this.tooltipRef = node; }}>
            <Header />
            <PageStats
              parentTabUrl={parentTabUrl}
              handleClick={this.handleClick}
            />
            <ThemedTabs
              passedSelectHandler={this.sendTrackerInformationTabMetrics}
              mountOnEnter
              tabs={[
                { title: 'Overview', view: <OverviewTab /> },
                { title: 'Global Trends', view: <GlobalTrendsTab /> },
                { title: 'Tracker Parentage', view: <TrackerParentageTab /> },
              ]}
            />
            <ThemedToast />
            <div className="Graph__tooltip" />
            <div className="Graph__tooltipCaret" />
          </div>
        </div>
      </React.Fragment>
    );
  }
}

App.propTypes = {
  parentTabUrl: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    setInitialTabInfo: PropTypes.func.isRequired,
    updatePageSize: PropTypes.func.isRequired,
    updateTrackerInfo: PropTypes.func.isRequired,
    updateTimingPerformance: PropTypes.func.isRequired,
    clearState: PropTypes.func.isRequired,
    updateParentTabUrl: PropTypes.func.isRequired,
    updateIsLive: PropTypes.func.isRequired,
    updateTooltipRef: PropTypes.func.isRequired,
  }).isRequired,
  messageCreators: PropTypes.shape({
    sendMetrics: PropTypes.func.isRequired,
  }).isRequired,
};

export default App;
