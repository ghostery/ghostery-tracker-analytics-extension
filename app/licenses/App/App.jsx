import React from 'react';
import Navbar from 'react-bootstrap/Navbar';
import '../../vendor/bootstrap.scss';

import License from './License';
import licenses from '../../../tools/licenses/licenses.json';

import './styles.scss';

const App = () => {
  const licensesArray = [];
  Object.keys(licenses).forEach((key) => {
    licensesArray.push(licenses[key]);
  });

  const list = licensesArray.map((license, index) => (
    <License
      key={license.name}
      license={license}
      isLastItem={index === licensesArray.length - 1}
    />
  ));

  return (
    <div className="Licenses">
      <div className="Licenses__header d-flex justify-content-between">
        <Navbar.Brand
          className="Licenses__logo"
          href="https://www.ghostery.com/products/"
          target="_blank"
          rel="noopener noreferrer"
        >
          <img
            alt="Insights by Ghostery Icon"
            src="/dist/images/logo-icons/insights-text-white.svg"
            width="93px"
            height="30px"
          />
        </Navbar.Brand>
        <span className="Licenses__title">
          Licenses
        </span>
      </div>
      <div className="Licenses__list">{ list }</div>
      <div className="Licenses__copyright">
        Ghostery© is a Cliqz company. Learn more about Cliqz. ©Cliqz International GmbH, a wholly-owned subsidiary of Cliqz GmbH. All rights reserved.
      </div>
    </div>
  );
};

export default App;
