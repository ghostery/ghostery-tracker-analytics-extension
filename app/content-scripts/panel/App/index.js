import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import App from './App.jsx';
import {
  setInitialTabInfo,
  updatePageSize,
  updateTrackerInfo,
  updateTimingPerformance,
  updateIsLive,
} from '../../../store/actions/PageInfoActions';
import {
  updateTooltipRef,
  openToast,
  closeToast,
  updateSearchResults,
  updateLoginStatus,
  updateEmailVerificationSent,
  updateRegisterStatus,
  updateResetPasswordStatus,
} from '../../../store/actions/UserInterfaceActions';
import {
  updateUserInfo,
  updateLocalSettings,
} from '../../../store/actions/SettingsActions';
import { sendMetrics } from '../../../store/messages/messageCreators';

const mapStateToProps = () => ({
  messageCreators: { sendMetrics },
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    setInitialTabInfo,
    updatePageSize,
    updateTrackerInfo,
    updateTimingPerformance,
    updateIsLive,
    updateTooltipRef,
    openToast,
    closeToast,
    updateSearchResults,
    updateUserInfo,
    updateLocalSettings,
    updateLoginStatus,
    updateEmailVerificationSent,
    updateRegisterStatus,
    updateResetPasswordStatus,
  }, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
