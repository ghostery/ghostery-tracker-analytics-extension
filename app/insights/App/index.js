import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import App from './App.jsx';
import {
  setInitialTabInfo,
  updatePageSize,
  updateTrackerInfo,
  updateTimingPerformance,
  updateTrackerParentage,
  clearState,
  updateParentTabUrl,
  updateIsPageNotScanned,
  updateParentTabClosed,
  updateIsLive,
} from '../../store/actions/PageInfoActions';
import {
  updateTooltipRef,
  openToast,
  closeToast,
  updateSearchResults,
  updateLoginStatus,
  updateEmailVerificationSent,
  updateRegisterStatus,
  updateResetPasswordStatus,
} from '../../store/actions/UserInterfaceActions';
import {
  updateUserInfo,
  updateLocalSettings,
} from '../../store/actions/SettingsActions';
import { sendMetrics } from '../../store/messages/messageCreators';

const mapStateToProps = state => ({
  parentTabUrl: state.PageInfoReducer.parentTabUrl,
  showToasts: state.SettingsReducer.localSettings.showToasts,
  messageCreators: { sendMetrics },
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    setInitialTabInfo,
    updatePageSize,
    updateTrackerInfo,
    updateTimingPerformance,
    updateTrackerParentage,
    clearState,
    updateParentTabUrl,
    updateIsPageNotScanned,
    updateParentTabClosed,
    updateIsLive,
    updateTooltipRef,
    openToast,
    closeToast,
    updateSearchResults,
    updateUserInfo,
    updateLocalSettings,
    updateLoginStatus,
    updateRegisterStatus,
    updateResetPasswordStatus,
    updateEmailVerificationSent,
  }, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
