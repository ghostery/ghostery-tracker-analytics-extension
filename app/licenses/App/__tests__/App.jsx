import React from 'react';
import { shallow } from 'enzyme';
import path from 'path';
import fakeProps from 'react-fake-props';

import App from '../App.jsx';

jest.mock('../../../../tools/licenses/licenses.json', () => ({
  license: {
    name: 'test',
    repository: 'test-repo',
    licenses: 'test-license',
    publisher: 'test-publisher',
    email: 'test@test.com',
    licenseText: 'test',
    url: 'tst.com',
  },
}), { virtual: true });

describe('app/licenses/App', () => {
  describe('Snapshot tests with react-test-renderer', () => {
    test('App entry point component should render', () => {
      const componentPath = path.join(__dirname, '../App.jsx');
      const props = fakeProps(componentPath);

      const component = shallow(<App {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
