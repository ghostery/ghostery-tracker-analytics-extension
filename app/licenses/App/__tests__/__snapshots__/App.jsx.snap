// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/licenses/App Snapshot tests with react-test-renderer App entry point component should render 1`] = `
<div
  className="Licenses"
>
  <div
    className="Licenses__header d-flex justify-content-between"
  >
    <NavbarBrand
      className="Licenses__logo"
      href="https://www.ghostery.com/products/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Insights by Ghostery Icon"
        height="30px"
        src="/dist/images/logo-icons/insights-text-white.svg"
        width="93px"
      />
    </NavbarBrand>
    <span
      className="Licenses__title"
    >
      Licenses
    </span>
  </div>
  <div
    className="Licenses__list"
  >
    <License
      isLastItem={true}
      key="test"
      license={
        Object {
          "email": "test@test.com",
          "licenseText": "test",
          "licenses": "test-license",
          "name": "test",
          "publisher": "test-publisher",
          "repository": "test-repo",
          "url": "tst.com",
        }
      }
    />
  </div>
  <div
    className="Licenses__copyright"
  >
    Ghostery© is a Cliqz company. Learn more about Cliqz. ©Cliqz International GmbH, a wholly-owned subsidiary of Cliqz GmbH. All rights reserved.
  </div>
</div>
`;
